File: 1423. Maximum Points You Can Obtain from Cards/Solution.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int windowSize = cardPoints.length - k;
        int allSum = 0, windowSum = 0;
        for (int i = 0; i < windowSize; i++) {
            windowSum += cardPoints[i];
            allSum += cardPoints[i];
        }

        int minSum = windowSum;
        for (int i = windowSize; i < cardPoints.length; i++) {
            windowSum += cardPoints[i] - cardPoints[i - windowSize];
            minSum = Math.min(minSum, windowSum);
            allSum += cardPoints[i];
        }
        return allSum - minSum;
    }
}

File: 1423. Maximum Points You Can Obtain from Cards/Solution.py
from typing import List


class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        window_size = len(cardPoints) - k
        window_sum= sum(cardPoints[:window_size])
        min_sum = window_sum
        for i in range(window_size, len(cardPoints)):
            window_sum += cardPoints[i] - cardPoints[i - window_size]
            min_sum = min(min_sum, window_sum)
        return sum(cardPoints) - min_sum
